qiskit/transpiler/passes/synthesis/unitary_synthesis.py

```diff
@@ -81,32 +127,116 @@ def run(self, dag: DAGCircuit) -> DAGCircuit:
+                   for requested basis.
+                2. pulse_optimize is True and natural_direction is True but a preferred
+                   gate direction can't be determined from the coupling map or the
+                   relative gate lengths.
         """
+
         euler_basis = _choose_euler_basis(self._basis_gates)
         kak_gate = _choose_kak_gate(self._basis_gates)
-
         decomposer1q, decomposer2q = None, None
         if euler_basis is not None:
             decomposer1q = one_qubit_decompose.OneQubitEulerDecomposer(euler_basis)
         if kak_gate is not None:
-            decomposer2q = TwoQubitBasisDecomposer(kak_gate, euler_basis=euler_basis)
-
-        for node in dag.named_nodes("unitary"):
+            decomposer2q = TwoQubitBasisDecomposer(
+                kak_gate, euler_basis=euler_basis, pulse_optimize=self._pulse_optimize
+            )
 
+        for node in dag.named_nodes(*self._synth_gates):
+            if self._basis_gates and node.name in self._basis_gates:
+                continue
             synth_dag = None
+            wires = None
             if len(node.qargs) == 1:
                 if decomposer1q is None:
                     continue
                 synth_dag = circuit_to_dag(decomposer1q._decompose(node.op.to_matrix()))
             elif len(node.qargs) == 2:
                 if decomposer2q is None:
                     continue
-                synth_dag = circuit_to_dag(
-                    decomposer2q(node.op.to_matrix(), basis_fidelity=self._approximation_degree)
-                )
+                synth_dag, wires = self._synth_natural_direction(node, dag, decomposer2q)
             else:
                 synth_dag = circuit_to_dag(isometry.Isometry(node.op.to_matrix(), 0, 0).definition)
 
-            dag.substitute_node_with_dag(node, synth_dag)
+            dag.substitute_node_with_dag(node, synth_dag, wires=wires)
 
         return dag
+
+    def _synth_natural_direction(self, node, dag, decomposer2q):
+        layout = self.property_set["layout"]
+        natural_direction = None
+        synth_direction = None
+        physical_gate_fidelity = None
+        wires = None
+        dag_qubit_index = {qubit: index for index, qubit in enumerate(dag.qubits)}
+        if self._natural_direction in {None, True} and layout and self._coupling_map:
+            neighbors0 = self._coupling_map.neighbors(dag_qubit_index[node.qargs[0]])
+            zero_one = dag_qubit_index[node.qargs[1]] in neighbors0
+            neighbors1 = self._coupling_map.neighbors(dag_qubit_index[node.qargs[1]])
+            one_zero = dag_qubit_index[node.qargs[0]] in neighbors1
+            if zero_one and not one_zero:
+                natural_direction = [0, 1]
+            if one_zero and not zero_one:
+                natural_direction = [1, 0]
+
+        elif self._natural_direction in {None, True} and layout and self._backend_props:
```

Comment:
Based on my comment below about adding the properties to the preset pass managers:

```suggestion
        if self._natural_direction in {None, True} and natural_direction is None and layout and self._backend_props:
```

Reply:
I'm confused why check for `natural_direction is None`? It should also work for `natural_direction is True` I think.

Reply:
my assumption from a quick look at the code that the `natural_direction` variable there is `None` or a list of qubits if the coupling map is non-symmetric for the qubits the gate operates on. For example it's `[0, 1]` or `[1, 0]` or `None` if it's not found. But, I could have read the code wrong, so I'll defer to @ewinston on what the right logic here.

Reply:
`natural_direction` takes the values `None`, [1, 0], or [0, 1]. Perhaps I should change the name of this or `self._natural_direction`, which takes the values `None`, True, False.